Apps/W1/Email - Outlook REST API/app/src/EmailOAuthClient.Codeunit.al

```diff
@@ -12,19 +12,25 @@ codeunit 4507 "Email - OAuth Client" implements "Email - OAuth Client"
     [NonDebuggable]
     procedure GetAccessToken(var AccessToken: Text)
     begin
-        TryGetAccessTokenInternal(AccessToken);
+        TryGetAccessTokenInternal(AccessToken, false);
     end;
 
     [NonDebuggable]
     procedure TryGetAccessToken(var AccessToken: Text): Boolean
     begin
-        exit(TryGetAccessTokenInternal(AccessToken));
+        exit(TryGetAccessToken(AccessToken, false));
+    end;
+
+    [NonDebuggable]
+    procedure TryGetAccessToken(var AccessToken: Text; Force: Boolean): Boolean
+    begin
+        exit(TryGetAccessTokenInternal(AccessToken, Force));
     end;
 
     // Interfaces do not support properties for the procedures, so using an internal function
     [TryFunction]
     [NonDebuggable]
-    local procedure TryGetAccessTokenInternal(var AccessToken: Text)
+    local procedure TryGetAccessTokenInternal(var AccessToken: Text; Force: Boolean)
```

Comment:
What is the purpose of the `Force` parameter? From what I can see it just allows to skip attempting to get the token from cache.
Is it a workaround to the problem with cache you mentioned in the PR's summary? If yes, I think we should find a better solution (e. g. using `SignInUsingAuthorizationCode`, see comment above).

Reply:
Yes, this is a workaround for the problem that I've mentioned.
I've tried to fix it as you suggested using the SignInUsingAuthorizationCode, but it opens a Window asking the Account, probably I'm missing something, could you please help me figure it out?
I think the problem is the fact that the cache is never deleted, then if you change the setup, the previous cache still exists allowing you to send an email with that one.

Reply:
Thanks!

Reply:
I think that it's ok, but from the UX perspective, I don't find it clear to the end-user.
Probably nobody will change the Tenant of the secret key. But I think think that the path to making it work is not clear.

Anyway, looking to the PR purpose, I've added the tenant field and made it work, I'll add Ideas for this purpose